market_env/env.py

```diff
@@ -31,40 +34,42 @@ def __init__(
         self.mempools = mempools
         self.blocks = blocks
 
+
 class User:
-    def __init__(self, user_id, env: ChainEnv, name: str):
+    def __init__(self, user_id, env: ChainEnv) -> None:
         self.user_id = user_id
         self.env = env
-        self.env.users[name] = self
 
     # users initiate transactions to a recipient
-    def create_transaction(self, transaction_id, amount, recipient):
+    def create_transaction(self, transaction_id, amount, recipient) -> "Transaction":
         transaction = Transaction(transaction_id, amount, recipient)
         return transaction
-    
+
     # users send transactions to mempool
-    def send_transaction(self, transaction):
+    def send_transaction(self, transaction) -> None:
         self.env.mempools.add_transaction(transaction)
 
+
 class Proposer:
-    def __init__(self, proposer_id):
+    def __init__(self, proposer_id) -> None:
         self.proposer_id = proposer_id
 
     # select the block with the highest pay
-    def select_block(self, block_id, builder_blocks):
+    def select_block(self, block_id, builder_blocks) -> "Block":
```

Comment:
Why quotation marks around the output type ? @TammyQAQ

Reply:
You can add this line 
https://github.com/xujiahuayz/auto-gov/blob/7db32f1986790852383971f97f59949fe066f223/market_env/env.py#L3